synapseclient/api/json_schema_services.py

```diff
@@ -207,15 +265,399 @@ async def get_json_schema_derived_keys(
+
+    Arguments:
+        organization_name: The name of the organization to retrieve
+        synapse_client: If not passed in and caching was not disabled by
+                       `Synapse.allow_client_caching(False)` this will use the last created
+                       instance from the Synapse class constructor
+
+    Returns:
+        An Organization object containing:
+        - id: The numeric identifier of the organization
+        - name: The organization name
+        - createdOn: Creation timestamp
+        - createdBy: ID of the user who created the organization
+
+        Object matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/schema/Organization.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    return await client.rest_get_async(
+        uri=f"/schema/organization?name={organization_name}"
+    )
+
+
+async def list_organizations(
+    *, synapse_client: Optional["Synapse"] = None
+) -> AsyncGenerator[Dict[str, Any], None]:
+    """
+    Generator to list all JSON schema organizations.
+
+    Retrieves a list of all organizations that are visible to the caller. This operation
+    does not require authentication and will return all publicly visible organizations.
+
+    Arguments:
+        synapse_client: If not passed in and caching was not disabled by
+                       `Synapse.allow_client_caching(False)` this will use the last created
+                       instance from the Synapse class constructor
+
+    Returns:
+        A list of Organization objects, each containing:
+        - id: The numeric identifier of the organization
+        - name: The organization name
+        - createdOn: Creation timestamp
+        - createdBy: ID of the user who created the organization
+
+        Object matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/schema/Organization.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    request_body = {}
+
+    async for item in rest_post_paginated_async(
+        "/schema/organization/list", body=request_body, synapse_client=client
+    ):
+        yield item
+
+
+def list_organizations_sync(
+    *, synapse_client: Optional["Synapse"] = None
+) -> Generator[Dict[str, Any], None, None]:
+    """
+    Generator to list all JSON schema organizations.
+
+    Retrieves a list of all organizations that are visible to the caller. This operation
+    does not require authentication and will return all publicly visible organizations.
+
+    Arguments:
+        synapse_client: If not passed in and caching was not disabled by
+                       `Synapse.allow_client_caching(False)` this will use the last created
+                       instance from the Synapse class constructor
+
+    Returns:
+        A list of Organization objects, each containing:
```

Comment:
Nit: since now the return object is a generator, the docstring here should be modified?